src/SFA.DAS.EmploymentCheck.Functions/Repositories/DataCollectionsResponseRepository.cs

```diff
@@ -95,5 +98,31 @@ public async Task<DataCollectionsResponse> Get(DataCollectionsResponse response)
 
             return await sqlConnection.GetAsync<DataCollectionsResponse>(response.ApprenticeEmploymentCheckId);
         }
+
+        public async Task<DataCollectionsResponse> GetByEmploymentCheckId(long apprenticeEmploymentCheckId)
+        {
+            DataCollectionsResponse response = null;
+            var dbConnection = new DbConnection();
+            await using (var sqlConnection = await dbConnection.CreateSqlConnection(
+                _connectionString,
+                _azureServiceTokenProvider)
+            )
+            {
+                await sqlConnection.OpenAsync();
+
+                var parameter = new DynamicParameters();
+                parameter.Add("@ApprenticeEmploymentCheckId", apprenticeEmploymentCheckId, DbType.Int64);
+
+                response = (await sqlConnection.QueryAsync<DataCollectionsResponse>(
+                    sql: "SELECT    TOP(1) * " +
+                         "FROM      [Cache].[DataCollectionsResponse] " +
+                         "WHERE     ApprenticeEmploymentCheckId = @ApprenticeEmploymentCheckId " +
+                         "ORDER BY  CreatedOn DESC ", // return the last one stored if there are multiple rows (we are retreiving the data by a non-key column so potentially their could be multiple matching rows)
```

Comment:
Don't need this comment, the code is self explanatory.

Reply:
Removed comment.